src/button-dropdown/internal.tsx

```diff
@@ -94,6 +94,12 @@ const InternalButtonDropdown = React.forwardRef(
       }
     }, [isOpen, wasOpen]);
 
+    useEffect(() => {
+      if (!wasOpen && isOpen && isInRestrictedView) {
```

Comment:
Why checking `wasOpen` and `isInRestrictedView`?

If it serves some special purposes it would be nice to have dedicated tests for it.

Reply:
removed `isInRestrictedView `

`wasOpen` is to make sure that the focus happens when the dropdown is previously closed. but after a bit of thinking it's seems trivial